frontend/pkg/frontend/middleware_tracing.go

```diff
@@ -16,3 +22,44 @@ import (
 func MiddlewareTracing(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
 	otelhttp.NewHandler(http.Handler(next), fmt.Sprintf("HTTP %s", r.Method)).ServeHTTP(w, r)
 }
+
+// ContextWithTraceCorrelationData adds correlationData as attributes to a propagated span.
+// It also registers a baggage with correlation data to the context.
+// If the context does not maintain a span, it has no effect.
+func ContextWithTraceCorrelationData(ctx context.Context, data *arm.CorrelationData) context.Context {
+	logger := LoggerFromContext(ctx)
+	// NOTE: Here the middleware span is extended by further attributes.
+	// If the tracingMiddleware is not registered, this lines will have no effect.
+	span := trace.SpanFromContext(ctx)
+	span.SetAttributes(attribute.String("correlation.id", data.CorrelationRequestID))
```

Comment:
suggestion: we could reduce duplication a bit.

````golang
    b, _ := bagagge.New() // will never err
    for _, := range []struct{
      name string
      value string
    }{
    } {
        {
            name: "correlation.id",
            value: data.CorrelationRequestID,
        },
        {
            ...
        },
        {
            ...
        },
    } {
        span.SetAttributes(attribute.String(name, value), value)
        m, err := baggage.NewMemberRaw(name, value)
        ...
        b.SetMember(m)
    }
````

Reply:
mh.. changed it :+1: